sympy/core/basic.py

```diff
@@ -1037,14 +1037,17 @@ def _eval_subs(self, old, new):
         """
         return None
 
-    def xreplace(self, rule):
+    def xreplace(self, rule, evaluate=True):
```

Comment:
Probably `evaluate` should be by default `False` here since it is just used for replacing the occurrences of objects and not anything more.

Reply:
I think I agree